module/zfs/zvol.c

```diff
@@ -1479,14 +1479,42 @@ zvol_rename_minor(zvol_state_t *zv, const char *newname)
 	set_disk_ro(zv->zv_disk, readonly);
 }
 
+typedef struct minors_job {
+	list_t *list;
+	list_node_t link;
+	/* input */
+	char *name;
+	/* output */
+	int error;
+} minors_job_t;
+
+static void
+zvol_prefetch_minors_impl(void *arg)
+{
+	int error;
+	minors_job_t *job = arg;
+	char *dsname = job->name;
+	objset_t *os = NULL;
+
+	error = dmu_objset_own(dsname, DMU_OST_ZVOL, B_TRUE, zvol_tag, &os);
+	if (error)
+		goto out;
+
+	dmu_prefetch(os, ZVOL_OBJ, 0, 0, 0, ZIO_PRIORITY_SYNC_READ);
+	dmu_objset_disown(os, zvol_tag);
+out:
```

Comment:
You could avoid the `goto` and potentially simplify  this by restructuring it as:

```
	job->error = dmu_objset_own(dsname, DMU_OST_ZVOL, B_TRUE, zvol_tag, &os);
	if (job->error == 0) {
		dmu_prefetch(os, ZVOL_OBJ, 0, 0, 0, ZIO_PRIORITY_SYNC_READ);
		dmu_objset_disown(os, zvol_tag);
	}
```